### PR TITLE
Fix high impact issues reported by Coverity

### PR DESCRIPTION
**Pull request by FernandoCP**
|Related issue|
|---|
|https://github.com/wazuh/wazuh/issues/4991|

## Description
Fixed resource leak in logcollector and "read/use after free" bugs in wazuh_db, that had been reported by Coverity.

## Tests

- Compilation without warnings in every supported platform
  - [X] Linux
  - [X] Windows
  - [x] MAC OS X
- [X] Source installation
- [X] Source upgrade

- [X] Test upgrade old db.

- Memory tests for Linux
  - [x] Scan-build report
  - [x] Coverity
  - [X] Valgrind (memcheck and descriptor leaks check)

- Memory tests for Windows
  - [x] Scan-build report
  - [ ] Coverity

- Memory tests for macOS
  - [x] Scan-build report

### CODE REVIEW

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 0c8383bf6669122fc8b966a847200c544dcd8ca5
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1658/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1658/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1658/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1658.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1658/Analysis_dynamic_valgrind-v3.13.0-B1658.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 0c8383bf6669122fc8b966a847200c544dcd8ca5
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1659/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1659/ossec.log)
**Coverage:** 

```json
Lines executed:26.47% of 3056

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1659/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1659.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1659/Analysis_dynamic_valgrind-v3.13.0-B1659.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** c3ca9e294ee6b3f4a39efc7098f5961e73c9a27a
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1660/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1660/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1660/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1660.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1660/Analysis_dynamic_valgrind-v3.13.0-B1660.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** c3ca9e294ee6b3f4a39efc7098f5961e73c9a27a
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1661/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1661/ossec.log)
**Coverage:** 

```json
Lines executed:26.51% of 3056

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1661/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1661.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1661/Analysis_dynamic_valgrind-v3.13.0-B1661.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** ae4854050e62a49f96dfe57814ffedb19c92f8c3
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1763/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1763/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1763/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1763.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1763/Analysis_dynamic_valgrind-v3.13.0-B1763.log)

File: src/wazuh_db/wdb_upgrade.c [hunk 1]
```diff
@@ -79,24 +78,26 @@ wdb_t * wdb_backup(wdb_t *wdb, int version) {
...
+                merror("Can't open SQLite backup database '%s': %s", path, sqlite3_errmsg((*wdb)->db));
+                sqlite3_close_v2((*wdb)->db);
                 free(sagent_id);
-                return NULL;
+                return -1;
             }
 
-            new_wdb = wdb_init(db, sagent_id);
+            *wdb = wdb_init(db, sagent_id);
+    	    free(sagent_id);
```
Comment (vikman90) [inline src/wazuh_db/wdb_upgrade.c:92]: The indentation seems not to be correct.

File: src/wazuh_db/wdb_upgrade.c [hunk 2]
```diff
@@ -32,9 +32,9 @@ wdb_t * wdb_upgrade(wdb_t *wdb) {
     char db_version[OS_SIZE_256 + 2];
     int version = 0;
 
-    switch (wdb_metadata_get_entry(wdb, "db_version", db_version)) {
+    switch (wdb_metadata_get_entry(*wdb, "db_version", db_version)) {
     case -1:
-        return wdb;
+        return 0;
```
Comment (vikman90) [inline src/wazuh_db/wdb_upgrade.c:37]: The return type is a boolean, please use `false` and `true`.

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 4024ab276a3132467089299955076df1278f425a
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1779/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1779/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1779/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1779.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1779/Analysis_dynamic_valgrind-v3.13.0-B1779.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 4024ab276a3132467089299955076df1278f425a
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1780/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1780/ossec.log)
**Coverage:** 

```json
Lines executed:27.00% of 3159

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1780/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1780.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1780/Analysis_dynamic_valgrind-v3.13.0-B1780.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 6f8013ef48de35ae4850786c2f88fa4fe03b7226
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1793/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1793/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1793/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1793.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1793/Analysis_dynamic_valgrind-v3.13.0-B1793.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 6f8013ef48de35ae4850786c2f88fa4fe03b7226
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1794/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1794/ossec.log)
**Coverage:** 

```json
Lines executed:27.00% of 3159

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1794/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1794.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1794/Analysis_dynamic_valgrind-v3.13.0-B1794.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 0d96a2b7628984d55718b43752ef33a78f815ce1
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1833/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1833/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1833/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1833.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1833/Analysis_dynamic_valgrind-v3.13.0-B1833.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 0d96a2b7628984d55718b43752ef33a78f815ce1
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1834/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1834/ossec.log)
**Coverage:** 

```json
Lines executed:27.07% of 3159

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1834/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1834.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1834/Analysis_dynamic_valgrind-v3.13.0-B1834.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 0d96a2b7628984d55718b43752ef33a78f815ce1
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1841/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1841/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1841/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1841.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1841/Analysis_dynamic_valgrind-v3.13.0-B1841.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 0d96a2b7628984d55718b43752ef33a78f815ce1
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1842/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1842/ossec.log)
**Coverage:** 

```json
Lines executed:27.00% of 3159

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1842/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1842.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1842/Analysis_dynamic_valgrind-v3.13.0-B1842.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind logcollector analysis:
**Commit:** 0d96a2b7628984d55718b43752ef33a78f815ce1
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1858/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1858/ossec.log)
**Coverage:** 

```json
Lines executed:31.41% of 1859

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1858/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1858.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1858/Analysis_dynamic_valgrind-v3.13.0-B1858.log)

**Comment by wazuhci** [comment]
## Analysis_dynamic_valgrind syscheckd analysis:
**Commit:** 0d96a2b7628984d55718b43752ef33a78f815ce1
**Parameters:** track-origins="yes" leak-check="full" keep-stacktraces="alloc-and-free" track-fds="yes" num-callers="20" show-leak-kinds="definite,indirect"
**Agent config:** [ossec.conf](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1857/ossec.conf)
**Agent logs:** [ossec.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1857/ossec.log)
**Coverage:** 

```json
Lines executed:27.00% of 3159

```
**Coverage report detail:** [coverage-report-summary](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1857/coverage-report-detail)
**Valgrind result:** [Analysis_dynamic_valgrind-v3.13.0-B1857.log](https://ci.wazuh.com/data/Analysis_dynamic_valgrind/v3.13.0/B1857/Analysis_dynamic_valgrind-v3.13.0-B1857.log)

**Pull request merged**